pkgs/top-level/all-packages.nix

```diff
@@ -17759,6 +17759,7 @@ with pkgs;
 
   libxml2 = callPackage ../development/libraries/libxml2 {
     python = python3;
+    inherit (darwin) libiconv;
```

Comment:
```suggestion
```

Reply:
This is fairly commonly used in `all-packages.nix` for cases where the dependency is only used on Darwin, I figured it's a little clearer this way rather than relying on the conditional/stubbed-out package (iconv is part of glibc). Maybe the other cases should be changed too though? I'm not aware of any particular standard one way or the other for this.

Reply:
I noticed a couple days ago, that we should maybe update the definition of libiconv: https://github.com/NixOS/nixpkgs/pull/138545#discussion_r713755466

Maybe something we can tackle in a separate PR

Reply:
Hm, I'm a little confused; shouldn't `!stdenv.isLinux` imply `stdenv.hostPlatform.libc != "glibc"` in almost all cases to begin with? What's incorrect about the current `libiconv`/`libintl` definitions?